test/groovy/DockerConfigTest.groovy

```diff
@@ -0,0 +1,92 @@
+
+import org.junit.Test
+import io.jenkins.infra.DockerConfig
+import io.jenkins.infra.InfraConfig
+
+import static org.junit.Assert.assertEquals
+import static org.junit.Assert.assertFalse
+import static org.junit.Assert.assertTrue
+
+class DockerConfigTest {
+
+    static String testImageName = "imagename"
+
+    @Test
+    void canHandleDefaultConfiguration() throws Exception {
+        def infraConfig = new InfraConfig(['JENKINS_URL':'https://ci.jenkins.io/'])
+        assertTrue(infraConfig.isRunningOnJenkinsInfra())
+        assertFalse(infraConfig.isInfra())
+        assertFalse(infraConfig.isTrusted())
+
+        def dockerConfig = new DockerConfig(testImageName, [:], infraConfig)
```

Comment:
Proposal: change the constructor signature to have `config` as last argument

For instance:

```
public DockerConfig(String imageName, InfraConfig infraConfig, Map config=[:]) {
```